R/hiper_glm.R

```
@@ -1,14 +1,21 @@
 #' @export
-hiper_glm <- function(design, outcome, model = 'linear') {
+hiper_glm <- function(design, outcome, model = 'linear', method = "BFGS") {
   supported_model <- c('linear', 'logit')
   if (! (model %in% supported_model)){
     stop(sprintf("The model %s is not supported.", model))
   }
-  warning("The function is yet to be implemented.")
-  # TODO: model fitting by MLE
-  hglm_out <- list()
-  class(hglm_out) <- 'hglm'
+  if (model == 'linear') {
+    hglm_out <- list()
+    class(hglm_out) <- 'hglm'
+    if (method == "pinv") {
+      hglm_out$coef <- lm_mle_pinv(design, outcome)
+    }
+    if (method == "BFGS") {
+      hglm_out$coef <- lm_mle_BFGS(design, outcome)
+    }
+  }
+  else {
+    warning("The function is yet to be implemented.")
```

-1

In this case you'd want to raise an error, not a warning. 

Also in general try to make warning/error messages more specific.  E.g. if I see this, I don't know if it's because i specified an invalid model, or an invalid method.